File: boosting/src/learning/regressors/WeightedRegressor.java
package learning.regressors;

import data.Data;
import data.RegressionData;

import java.util.List;

public class WeightedRegressor implements Regressor {
    private final Regressor regressor;
    private double weight = 1;

    public WeightedRegressor(Regressor regressor) {
        this.regressor = regressor;
    }

    @Override
    public void train(List<RegressionData> data) {
        regressor.train(data);
    }

    @Override
    public double regress(Data data) {
        return weight * regressor.regress(data);
    }

    public void setWeight(double weight) {
        this.weight = weight;
    }
}

File: boosting/src/learning/regressors/WeightedRegressor.kt
package learning.regressors

import data.Data
import data.DataWithResult
import learning.model.ModelWithTeacher

class WeightedRegressor(private val regressor: ModelWithTeacher) : ModelWithTeacher {

    private var weight = 1.0

    override fun train(data: List<DataWithResult>) {
        regressor.train(data)
    }

    override fun output(data: Data): Double {
        return weight * regressor.output(data).toDouble()
    }

    fun setWeight(weight: Double) {
        this.weight = weight
    }
}
